### ISSUE TITLE
Can't figure out how to set up the proxy with Android phone

### ISSUE THREAD

**Issue by stoefln** [joeferner/node-http-mitm-proxy#272]
My goal is to route and record the traffic from an Android phone (Chrome or my own app) over the proxy.
After configuring my phone's wifi settings (IP and port of desktop machine), and trying to load a page in the mobile chrome, I am now facing a "Your connection is not private" dialog.
On the desktop my proxy displays following:
```
HTTPS_CLIENT_ERROR on : [Error: 8190125312:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 46
] {
  library: 'SSL routines',
  function: 'ssl3_read_bytes',
  reason: 'sslv3 alert certificate unknown',
  code: 'ERR_SSL_SSLV3_ALERT_CERTIFICATE_UNKNOWN'
}
HTTPS_CLIENT_ERROR on : [Error: 8190125312:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 46
] {
  library: 'SSL routines',
  function: 'ssl3_read_bytes',
  reason: 'sslv3 alert certificate unknown',
  code: 'ERR_SSL_SSLV3_ALERT_CERTIFICATE_UNKNOWN'
}
HTTPS_CLIENT_ERROR on : [Error: 8190125312:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 46
] {
  library: 'SSL routines',
  function: 'ssl3_read_bytes',
  reason: 'sslv3 alert certificate unknown',
  code: 'ERR_SSL_SSLV3_ALERT_CERTIFICATE_UNKNOWN'
}
HTTPS_CLIENT_ERROR on : [Error: 8190125312:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 46
] {
  library: 'SSL routines',
  function: 'ssl3_read_bytes',
  reason: 'sslv3 alert certificate unknown',
  code: 'ERR_SSL_SSLV3_ALERT_CERTIFICATE_UNKNOWN'
}
HTTPS_CLIENT_ERROR on : [Error: 8190125312:error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 46
] {
  library: 'SSL routines',
  function: 'ssl3_read_bytes',
  reason: 'sslv3 alert certificate unknown',
  code: 'ERR_SSL_SSLV3_ALERT_CERTIFICATE_UNKNOWN'
}
```

I guess this is somehow related to certificates, but can someone point me into the right direction of how to solve this issue?

**Comment by stoefln** [comment]
Just found this: https://github.com/joeferner/node-http-mitm-proxy/issues/76
Seems very much related.

**Comment by stoefln** [comment]
This is how it's done for Charles Proxy: https://stackoverflow.com/a/39227880/457059
I am just clueless about which certificate to install on the Android side in case of using node-http-mitm-proxy

**Comment by stoefln** [comment]
Found the solution, just in case somebody gets stuck at the same point:

node-http-mitm-proxy creates the certificates itself. You just need to install the ca.pem certificate on the target device - that's it.

To do that, just copy (adb push) the .http-mitm-proxy/certs/ca.pem to your target device. To install it, you best follow this guide: https://docs.familyzone.com/help/how-to-install-certificate-authority-android

**Comment by TonyxSun** [comment]
hi @stoefln, I am running into the same issue, except even after adding ca.pem and installing it in setttings, I continue to get the same errors. 

Looking online, I note that after Android 7, ["Apps that target API Level 24 and above no longer trust user or admin-added CAs for secure connections, by default."](https://android-developers.googleblog.com/2016/07/changes-to-trusted-certificate.html)

Did you have to deal with the same issue?

**Comment by Thibobbs** [comment]
Hello everyone,

I'm currently trying to route and record the traffic from my Android phone. Unfortunately, I'm encountering an issue where my phone is unable to connect to the proxy. Specifically, I'm receiving the error message: ERR_PROXY_CONNECTION_FAILED (on Chrome), indicating that the proxy cannot be reached. I've experimented with other proxy libraries and encountered a different error message, suggesting that the proxy is being detected.

Have any of you configured this proxy to work specifically with Android devices?

Here is my code (can't be simpler than that):
```
import { Proxy } from "http-mitm-proxy";
const proxy = new Proxy();
proxy.listen({
  port: 8081,
});
```


EDIT:
I finally found the solution to this problem, I just needed to add the host property to the proxy parameters.
Here is the fixed version of the code:
```
import { Proxy } from "http-mitm-proxy";
const proxy = new Proxy();
proxy.listen({
  port: 8081,
  host: '0.0.0.0',
});
```
